mycroft/skills/core.py

```diff
@@ -231,13 +233,41 @@ def __init__(self, name=None, bus=None):
         self.skill_id = ''  # will be set from the path, so guaranteed unique
         self.voc_match_cache = {}
 
+    @property
+    def enclosure(self):
+        if self._enclosure:
+            return self._enclosure
+        else:
+            LOG.error("ERROR:  Skill not fully initialized.  Move code from " +
+                      " __init__() to initialize() to correct this.")
+            tb = "Traceback:\n"
+            for line in traceback.format_stack()[:-1]:
+                if line.strip():
+                    tb += line
+            LOG.error(tb)
```

Comment:
What is the benefit of this over using `LOG.exception()` to log the traceback?

Reply:
This isn't happening inside an exception -- I'm just dumping the stack trace to show the skill author where they goofed up.  I also strip out a level of the call stack so it is more obvious.

I suppose I could throw an exception instead, but this really enforces that they should never be accessing self.bus or self.enclosure until AFTER the __init__() has completed.  This will always dump text into the logs even if they do gracefully handle the self.bus being None.

Reply:
Ok, then I get the intent of this.